### ISSUE TITLE
Dataloader values do not match with original dataframe when target_normalizer=None

### ISSUE THREAD

**Issue by AjinkyaBankar** [jdb78/pytorch-forecasting#1477]
- PyTorch-Forecasting version: 1.0.0
- PyTorch version: 2.1.0
- Python version: 3.9
- Operating System: Linux

### Expected behavior

I executed code in order to create and check train and val dataloaders by assuming `target_normalizer=None`. I wanted to see the data formatted as sliding window of independent variables and target. Since the target is not normalized, I expected to see original values of the 'target'.

### Actual behavior

However, result was fine for series=0 and series=1. But there is mismatch of values in the dataframe and dataloaders from series=2 onwards. Also, if `n_series` is a small number (e.g., 5) then all the series values of 'target' match with dataloaders.

### Code to reproduce the problem

```
import lightning.pytorch as pl
from lightning.pytorch.callbacks import EarlyStopping
import matplotlib.pyplot as plt
import pandas as pd
import torch

from pytorch_forecasting import Baseline, DeepAR, TimeSeriesDataSet
from pytorch_forecasting.data import NaNLabelEncoder
from pytorch_forecasting.data.examples import generate_ar_data
from pytorch_forecasting.metrics import MAE, SMAPE, MultivariateNormalDistributionLoss

data = generate_ar_data(seasonality=10.0, timesteps=100, n_series=50, seed=42)
data["static"] = 2
data["date"] = pd.Timestamp("2020-01-01") + pd.to_timedelta(data.time_idx, "D")
data.head()

data = data.astype(dict(series=str))

# create dataset and dataloaders
max_encoder_length = 10
max_prediction_length = 1

training_cutoff = data["time_idx"].max() - max_prediction_length

context_length = max_encoder_length
prediction_length = max_prediction_length

training = TimeSeriesDataSet(
    data[lambda x: x.time_idx <= training_cutoff],
    time_idx="time_idx",
    target="value",
    categorical_encoders={"series": NaNLabelEncoder().fit(data.series)},
    group_ids=["series"],
    static_categoricals=[
        "series"
    ],  # as we plan to forecast correlations, it is important to use series characteristics (e.g. a series identifier)
    time_varying_unknown_reals=["value"],
    max_encoder_length=context_length,
    max_prediction_length=prediction_length,
    target_normalizer=None
)

validation = TimeSeriesDataSet.from_dataset(training, data, min_prediction_idx=training_cutoff + 1)
batch_size = 4
# synchronize samples in each batch over time - only necessary for DeepVAR, not for DeepAR
train_dataloader = training.to_dataloader(
    train=True, batch_size=batch_size, num_workers=0, batch_sampler="synchronized"
)
val_dataloader = validation.to_dataloader(
    train=False, batch_size=batch_size, num_workers=0, batch_sampler="synchronized"
)

# Print the dataolader values
for x, (y, weight) in iter(train_dataloader):
    print('X')
    print(x)
    print('\n')
    print('y')
    print(y)
    print('---------------------------------')
```
![image](https://github.com/jdb78/pytorch-forecasting/assets/77496039/c660d126-80b1-43bc-ae70-5d8fdff53aad)
![image](https://github.com/jdb78/pytorch-forecasting/assets/77496039/dc03d5a1-26ce-448f-a5fd-39150564eb9d)
![image](https://github.com/jdb78/pytorch-forecasting/assets/77496039/c369319a-9a76-4968-9afa-48dc006ab083)
![image](https://github.com/jdb78/pytorch-forecasting/assets/77496039/2196e979-7263-469c-91bb-45d1dbbc34e9)

```
The target value of series 4 at time_idx=17 does not match with that in dataframe.
```
![image](https://github.com/jdb78/pytorch-forecasting/assets/77496039/f46568e6-36e4-4b83-a6d1-895dcc21aab3)

**Comment by AjinkyaBankar** [comment]
I was able to relate them with `training.x_to_index(x)` method.

**Comment by AjinkyaBankar** [comment]
I was able to relate them with `training.x_to_index(x)` method.
